Mathlib/Topology/Separation.lean

```diff
@@ -2100,6 +2167,31 @@ lemma SeparatedNhds.of_isCompact_isClosed {s t : Set X}
 @[deprecated (since := "2024-01-28")]
 alias separatedNhds_of_isCompact_isClosed := SeparatedNhds.of_isCompact_isClosed
 
+/-- This technique to witness `HasSeparatingCover` in regular Lindelöf topological spaces
+will be used to prove regular Lindelöf spaces are normal. -/
+lemma IsClosed.HasSeparatingCover {s t : Set X} [r: RegularSpace X] [LindelofSpace X]
+    (s_cl : IsClosed s) (t_cl : IsClosed t) (st_dis : Disjoint s t) : HasSeparatingCover s t := by
+  -- `IsLindelof.indexed_countable_subcover` requires the space be Nonempty
+  rcases isEmpty_or_nonempty X with empty_X | nonempty_X
+  · rw [subset_eq_empty (t := s) (fun ⦃_⦄ _ ↦ trivial) (univ_eq_empty_iff.mpr empty_X)]
+    exact has_separating_covers_iff_separated_nhds.mpr (SeparatedNhds.empty_left t) |>.1
+  -- This is almost `HasSeparatingCover`, but is not countable. We define for all `a : X` for use
+  -- with `IsLindelof.indexed_countable_subcover` momentarily.
+  have (a : X) : ∃ n : Set X, IsOpen n ∧ Disjoint (closure n) t ∧ (a ∈ s → a ∈ n) := by
+    wlog ains : a ∈ s
+    · exact ⟨∅, isOpen_empty, SeparatedNhds.empty_left t |>.disjoint_closure_left, fun a ↦ ains a⟩
+    obtain ⟨n, nna, ncl, nsubkc⟩ := (((regularSpace_TFAE X).out 0 3).mp r:) a tᶜ
+      (t_cl.compl_mem_nhds (disjoint_left.mp st_dis ains))
+    exact ⟨interior n, isOpen_interior, disjoint_left.mpr fun ⦃a⦄ ain ↦
+      nsubkc ((IsClosed.closure_subset_iff ncl).mpr interior_subset ain),
+      fun _ ↦ mem_interior_iff_mem_nhds.mpr nna⟩
+  -- By Lindelöf, we may obtain a countable subcover witnessing `HasSeparatingCover`
+  choose u u_open u_dis u_nhd using this
+  obtain ⟨f, f_cov⟩ := s_cl.isLindelof.indexed_countable_subcover
+    u u_open (fun a ainh ↦ mem_iUnion.mpr ⟨a, u_nhd a ainh⟩)
+  exact ⟨u ∘ f, ⟨f_cov, fun n ↦ ⟨u_open (f n), u_dis (f n)⟩⟩⟩
```

Comment:
The second level of ⟨⟩ nesting should not be necessary.

Reply:
3a2386b